test-sol/actions/sky/SkyStakingEngineClaimRewards.t.sol

```diff
@@ -0,0 +1,100 @@
+    //////////////////////////////////////////////////////////////////////////*/
+    SmartWallet wallet;
+    address walletAddr;
+    address sender;
+    uint256 constant AMOUNT = 1000e18;
+    address constant USDS_FARM = 0x38E4254bD82ED5Ee97CD1C4278FAae748d998865;
+    address constant SPARK_FARM = 0x99cBC0e4E6427F6939536eD24d1275B95ff77404;
+
+    SkyStakingEngineOpen open;
+    SkyStakingEngineStake stake;
+    SkyStakingEngineSelectFarm selectFarm;
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                  SETUP FUNCTION
+    //////////////////////////////////////////////////////////////////////////*/
+    function setUp() public override {
+        forkMainnetLatest();
+
+        wallet = new SmartWallet(bob);
+        sender = wallet.owner();
+        walletAddr = wallet.walletAddr();
+
+        cut = new SkyStakingEngineClaimRewards();
+        open = new SkyStakingEngineOpen();
+        stake = new SkyStakingEngineStake();
+        selectFarm = new SkyStakingEngineSelectFarm();
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                     TESTS
+    //////////////////////////////////////////////////////////////////////////*/
+    function test_skyStakingEngineClaimRewards_Direct_USDS_FARM() public {
+        _baseTest(true, USDS_FARM, USDS_ADDRESS);
+    }
+
+    function test_skyStakingEngineClaimRewards_USDS_FARM() public {
+        _baseTest(false, USDS_FARM, USDS_ADDRESS);
+    }
+
+    function test_skyStakingEngineClaimRewards_Direct_SPARK_FARM() public {
+        _baseTest(true, SPARK_FARM, SPARK_ADDRESS);
+    }
+
+    function test_skyStakingEngineClaimRewards_SPARK_FARM() public {
+        _baseTest(false, SPARK_FARM, SPARK_ADDRESS);
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                     HELPERS
+    //////////////////////////////////////////////////////////////////////////*/
+
+    function _baseTest(bool _isDirect, address _farm, address _rewardToken) internal {
+        //  Give SKY to sender and approve wallet
+        give(SKY_ADDRESS, sender, AMOUNT);
+        approveAsSender(sender, SKY_ADDRESS, walletAddr, AMOUNT);
+        uint256 index = 0;
+
+        //  Stake first
+        executeSkyStakingEngineStake(STAKING_ENGINE, index, _farm, AMOUNT, sender, open, selectFarm, stake, wallet);
+
+        skip(365 days);
+
+        //  Execution logic of claiming rewards
+        bytes memory executeActionCallData =
+            executeActionCalldata(skyStakingEngineClaimRewardsEncode(STAKING_ENGINE, index, _farm, sender), _isDirect);
+        vm.expectEmit(true, true, true, false, address(STAKING_ENGINE));
+        emit ILockstakeEngine.GetReward(walletAddr, index, _farm, sender, 0);
+        wallet.execute(address(cut), executeActionCallData, 0);
+
+        uint256 amount = IERC20(_rewardToken).balanceOf(sender);
```

Comment:
Change before and after balance and the difference, rather than just after balance